### DISCUSSION TITLE
Propagate baggage attributes as span attributes to root span

### DISCUSSION THREAD

**Discussion by MadhusudanN** [open-telemetry/opentelemetry-java-instrumentation#9563] [Q&A]
Hi,
 As explained in the below discussion forums ::
https://github.com/open-telemetry/opentelemetry-java/discussions/5834
https://github.com/open-telemetry/opentelemetry-java/discussions/4801

We need to create baggage with attributes in the current context  and propagate it to the downstream code enclosed within the current context. Use the custom span processor to extract from baggage and add as span attributes. This has worked fine.

Requirement ::
 Is there a way to set span attributes of root span with baggage attributes

Scenario ::
 A java based micro-service which has logging filter, baggage is set at this filter with some attributes and is propagated to further spans. 
The custom attributes are seen only at the controller layer and further but not at the root span. Please find the image below which depicts this.

Water fall diagram of trace :: 
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26345035/861bce7c-08fe-4d15-933e-4bf7b63114d5)

Custom attribute is the **user_id** which is  seen at controller layer ::

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26345035/08185dbc-1db6-4319-899c-82a65d28df1b)

Custom attribute **user_id** not seen at the root span ::
The root span is highlighted below 

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26345035/1d3216ed-30b0-4a3d-830a-9a51f1e13d41)

The root span attributes are shown below ::
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26345035/f3565f11-b2b6-4160-9f29-dcc785f6fe79)

In the above we do not see the user_id attribute.
Can someone please let me know if there is a way to propagate the baggage as span attribute at this root span level.

**Comment by laurit** [comment]
If the root span is created in the same service then you can get it with `Span root = LocalRootSpan.current();` and then just call `setAttribute` on the span. See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/LocalRootSpan.java If the root span is created by a remote service then there is no way to do this with the agent.
